src/main/java/me/lemire/longcompression/LongVariableByte.java

```diff
@@ -0,0 +1,343 @@
+                buf.put((byte) (extract7bitsmaskless(3, (val)) | (1 << 7)));
+            } else if (val >= 0 && val < (1L << 35)) {
+                buf.put((byte) extract7bits(0, val));
+                buf.put((byte) extract7bits(1, val));
+                buf.put((byte) extract7bits(2, val));
+                buf.put((byte) extract7bits(3, val));
+                buf.put((byte) (extract7bitsmaskless(4, (val)) | (1 << 7)));
+            } else if (val >= 0 && val < (1L << 42)) {
+                buf.put((byte) extract7bits(0, val));
+                buf.put((byte) extract7bits(1, val));
+                buf.put((byte) extract7bits(2, val));
+                buf.put((byte) extract7bits(3, val));
+                buf.put((byte) extract7bits(4, val));
+                buf.put((byte) (extract7bitsmaskless(5, (val)) | (1 << 7)));
+            } else if (val >= 0 && val < (1L << 49)) {
+                buf.put((byte) extract7bits(0, val));
+                buf.put((byte) extract7bits(1, val));
+                buf.put((byte) extract7bits(2, val));
+                buf.put((byte) extract7bits(3, val));
+                buf.put((byte) extract7bits(4, val));
+                buf.put((byte) extract7bits(5, val));
+                buf.put((byte) (extract7bitsmaskless(6, (val)) | (1 << 7)));
+            } else if (val >= 0 && val < (1L << 56)) {
+                buf.put((byte) extract7bits(0, val));
+                buf.put((byte) extract7bits(1, val));
+                buf.put((byte) extract7bits(2, val));
+                buf.put((byte) extract7bits(3, val));
+                buf.put((byte) extract7bits(4, val));
+                buf.put((byte) extract7bits(5, val));
+                buf.put((byte) extract7bits(6, val));
+                buf.put((byte) (extract7bitsmaskless(7, (val)) | (1 << 7)));
+            } else if (val >= 0 && val < (1L << 63)) {
+                buf.put((byte) extract7bits(0, val));
+                buf.put((byte) extract7bits(1, val));
+                buf.put((byte) extract7bits(2, val));
+                buf.put((byte) extract7bits(3, val));
+                buf.put((byte) extract7bits(4, val));
+                buf.put((byte) extract7bits(5, val));
+                buf.put((byte) extract7bits(6, val));
+                buf.put((byte) extract7bits(7, val));
+                buf.put((byte) (extract7bitsmaskless(8, (val)) | (1 << 7)));
+            } else {
+                buf.put((byte) extract7bits(0, val));
+                buf.put((byte) extract7bits(1, val));
+                buf.put((byte) extract7bits(2, val));
+                buf.put((byte) extract7bits(3, val));
+                buf.put((byte) extract7bits(4, val));
+                buf.put((byte) extract7bits(5, val));
+                buf.put((byte) extract7bits(6, val));
+                buf.put((byte) extract7bits(7, val));
+                buf.put((byte) extract7bits(8, val));
+                buf.put((byte) (extract7bitsmaskless(9, (val)) | (1 << 7)));
```

Comment:
I would have needed more time on this: the last byte is used for a single bit (7*9 == 63, this hold the 64-th bit). We may prefer relying on 9 bytes, where the last one does not rely on 7bits, but the full 8-bits. (I can reword if I'm not clear)

Reply:
I think what you did is fine. It is true that you can infer that the 9th byte in a sequence must be the final one... However,  the largest value of a long in Java is 9,223,372,036,854,775,807. So the the 10th byte is only used for negative longs.

Reply:
7*8 (VariableByte in 8 bytes holding 7-bits of data)+8 as last byte = 64b. I see no reason (yet) this idea not to work, and to store the worst-case scenario in 9 bytes instead of 10. I'll make a PR to confirm and propose this change.

Reply:
Yes, what you suggest will work. My point was that what you did is also reasonable.